core/config/engine.cpp

```diff
@@ -240,7 +240,16 @@ bool Engine::is_abort_on_gpu_errors_enabled() const {
 }
 
 int32_t Engine::get_gpu_index() const {
-	return gpu_idx;
+	//check for cmd argument override, otherwise get index by config
+	if (gpu_idx != -1)
+		return gpu_idx;
+	else {
+		int gpu_index_from_config = GLOBAL_GET("rendering/rendering_device/gpu_index");
+		if (gpu_index_from_config == -1)
+			return gpu_idx;
+		else
+			return gpu_index_from_config;
+	}
```

Comment:
Please check the [Code style guidelines](https://docs.godotengine.org/en/stable/contributing/development/code_style_guidelines.html) :slightly_smiling_face: 

I recommend setting up clang-format locally as described on that page.